builtin/credential/ldap/backend.go

```diff
@@ -106,7 +68,12 @@ func (b *backend) Login(ctx context.Context, req *logical.Request, username stri
 		return nil, logical.ErrorResponse("ldap backend not configured"), nil, nil
 	}
 
-	c, err := cfg.DialLDAP()
+	ldapClient := ldaputil.Client{
+		Logger: b.Logger(),
+		LDAP:   ldaputil.NewLDAP(),
+	}
+
+	c, err := ldapClient.DialLDAP(cfg)
```

Comment:
DialLDAP is exactly the same method, except it's being executed as part of the client instead as part of the config. I made a design choice that anything that makes an external call would be part of the ldap client.

Reply:
👍